lib/pin_screen/pin_password_input_field.dart

```diff
@@ -1,27 +1,75 @@
 import 'package:flutter/material.dart';
+import 'package:mood_tracker/pin_screen/provider/pin_notifier.dart';
+import 'package:provider/provider.dart';
 
 class PasswordInputField extends StatefulWidget {
-  PasswordInputField({required this.pin});
-
   var pin;
 
+  PasswordInputField({required this.pin});
+
   @override
   State<PasswordInputField> createState() => _PasswordInputFieldState();
 }
 
 class _PasswordInputFieldState extends State<PasswordInputField> {
   @override
   Widget build(BuildContext context) {
+    bool wrongPin = context.watch<PinNotifier>().wrongPin;
+    var color = context.watch<PinNotifier>().color;
+    // var changeColor = context.watch()<PinNotifier>().colorButton();
+
+    // bool pin = context.watch<PinNotifier>().pin;
+
+    // var color;
+
+    void colorButton() {
```

Comment:
move this function to class level and add encapsulation (make this function private)